internal/pkg/platform/platform_matcher.go

```diff
@@ -115,11 +115,22 @@ func getCPUVariant(os string, arch string) string {
 	return ""
 }
 
+// compatibility contains, for a specified architecture, a list of known variants, in the
+// order from most capable (most restrictive) to least capable (most compatible).
+// Architectures that don’t have variants should not have an entry here.
 var compatibility = map[string][]string{
 	"arm":   {"v7", "v6", "v5"},
 	"arm64": {"v8"},
 }
 
+// baseVariants contains, for a specified architecture, a variant that is known to be
+// supported by _all_ machines using that architecture.
+// Architectures that don’t have variants, or where there are possible versions without
+// an established variant name, should not have an entry here.
+var baseVariants = map[string]string{
+	"arm64": "v8",
```

Comment:
arm/v7 does not run all 32 bit arm variants?

Reply:
`baseVariants` answers a different question: “when we want an image for ARM/unknown, which variant is certainly acceptable?” Assuming v5 is a subset of v7, arm/v5 would be a better answer to that question than v7. I’m not sure whether v5 is a safe default — looking at `getCPUVariantArm`, values up to v3 seem to exist.

Hum, looking at docker://busybox:1.27.2 , we might need a `baseVariants` entry for ARM — but I really don’t know whether v5 is OK.

I’d be happy to hear from experts.

Reply:
I would say that it should take the lowest variant (most likely to work), but prefer a matching "arm/unkown" image if present (perfect match).

Or if it's not too incompatible, erroring out would probably also work, as except for the case where an "arm/unknown" image is requested explicitly (e.g. for mirroring) that shouldn't be happening at all.

Reply:
> I would say that it should take the lowest variant (most likely to work), but prefer a matching "arm/unkown" image if present (perfect match).

That’s what the code does, but which is the lowest variant for 32-bit ARM (on container-capable Linux systems, I guess)?

Reply:
The Linux kernel currently supports anything down to v4 AFAICT, though I've never seen container images for that...

Reply:
This is all for the “non-ARM executable wants an ARM image” case, so I think at least for now (while we need to urgently fix CRI-O) it might be good enough to require the caller to explicitly specify a variant.